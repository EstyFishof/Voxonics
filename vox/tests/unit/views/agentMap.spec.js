import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import AgentsMap from '@/views/Dashboard/AgentsMap.vue';
import VueTippy, { TippyComponent } from "vue-tippy";

const localVue = createLocalVue();
localVue.use(Vuex);
const router = new VueRouter();
localVue.use(Vuex);
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);

describe('support', () => {
    window.env = {
        WS_URL: 'ws://168.119.92.172:9595'
    };
    const store = new Vuex.Store({
        actions: {
            pushLogout({commit}, value) {},
            emitError({commit}, value) {},
        },
        getters: {
            userInfo: () => {}
        }
    });

    const wrapper = shallowMount(AgentsMap, { localVue, router, store });
    it('it should  Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('getTime toHaveBeenCalled', async () => {
        await wrapper.setData({data: [{time: 0}]});
        const spy = jest.spyOn(wrapper.vm, 'getTime');
        await wrapper.vm.getTime();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('computedTime toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'computedTime');
        await wrapper.vm.computedTime(0);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('switchViews toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'switchViews');
        await wrapper.vm.switchViews();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.isList).toStrictEqual(true);
        spy.mockRestore();
    });
    it('hideListOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'hideListOptions');
        await wrapper.vm.hideListOptions();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.isShowMenu).toStrictEqual('');
        expect(wrapper.vm.sortByShow).toStrictEqual(false);
        spy.mockRestore();
    });
    it('hideListOptions toHaveBeenCalled', async () => {
        await wrapper.setData({indexAgent: 'true'});
        const spy = jest.spyOn(wrapper.vm, 'hideListOptions');
        await wrapper.vm.hideListOptions();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('sort toHaveBeenCalled', async () => {
        await wrapper.setData({data: [{name: 'AAA'}]});
        const spy = jest.spyOn(wrapper.vm, 'sort');
        await wrapper.vm.sort();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('editMap toHaveBeenCalled', async () => {
        await wrapper.setData({data: [{name: 'AAA'}]});
        const spy = jest.spyOn(wrapper.vm, 'editMap');
        await wrapper.vm.editMap();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.initialData).toStrictEqual([{name: 'AAA'}]);
        spy.mockRestore();
    });
    it('addCard toHaveBeenCalled', async () => {
        await wrapper.setData({data: [{name: 'AAA'}]});
        const spy = jest.spyOn(wrapper.vm, 'addCard');
        await wrapper.vm.addCard();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('toBackMap toHaveBeenCalled', async () => {
        await wrapper.setData({initialData: [{name: 'AAA'}]});
        const spy = jest.spyOn(wrapper.vm, 'toBackMap');
        await wrapper.vm.toBackMap();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.data).toStrictEqual([{name: 'AAA'}]);
        spy.mockRestore();
    });
    it('deleteCard toHaveBeenCalled', async () => {
        await wrapper.setData({data: [{name: 'AAA', id: 111}]});
        const spy = jest.spyOn(wrapper.vm, 'deleteCard');
        await wrapper.vm.deleteCard({name: 'AAA', id: 111});
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.data).toStrictEqual([]);
        spy.mockRestore();
    });
    it('setPause toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setPause');
        await wrapper.vm.setPause({name: 'AAA', id: 111, onPause: true});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('logOutUser toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'logOutUser');
        await wrapper.vm.logOutUser({name: 'AAA', id: 111, onShift: true});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('updateMap toHaveBeenCalled', async () => {
        await wrapper.setData({data: [{name: 'AAA', id: 111}]});
        const spy = jest.spyOn(wrapper.vm, 'updateMap');
        await wrapper.setData({data: [{name: 'AAA', id: 111, onShift: true}]});
        await wrapper.setData({initialData: [{name: 'AAA', id: 111, onShift: true}]});
        await wrapper.vm.updateMap();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.initialData).toStrictEqual([]);
        spy.mockRestore();
    });
    it('showMenu toHaveBeenCalled', async () => {
        await wrapper.setData({isShowMenu: 111});
        const spy = jest.spyOn(wrapper.vm, 'showMenu');
        await wrapper.vm.showMenu({uid: 111});
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.isShowMenu).toStrictEqual('');
        spy.mockRestore();
    });
    it('showMenu toHaveBeenCalled', async () => {
        await wrapper.setData({isShowMenu: 111});
        const spy = jest.spyOn(wrapper.vm, 'showMenu');
        await wrapper.vm.showMenu({uid: 123});
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.isShowMenu).toStrictEqual(123);
        spy.mockRestore();
    });
    it('insert1 toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'insert1');
        await wrapper.vm.insert1({index: 1, data: {}});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('remove toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'remove');
        await wrapper.vm.remove(['aaa', 'bbb'], 'aaa');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("online");
        expect(spy).toHaveReturnedWith({ color: '#00C82C', icon: wrapper.vm.OnlineIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("paused");
        expect(spy).toHaveReturnedWith({ color: '#FFB200', icon: wrapper.vm.PauseIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("offline");
        expect(spy).toHaveReturnedWith({ color: '#FF3A29', icon: wrapper.vm.OfflineIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("calling");
        expect(spy).toHaveReturnedWith({ color: '#78ABCE', icon: wrapper.vm.CallingIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("ringing");
        expect(spy).toHaveReturnedWith({ color: '#78ABCE', icon: wrapper.vm.CallingIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("talking");
        expect(spy).toHaveReturnedWith({ color: '#276899', icon: wrapper.vm.TalkingIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor("blocked");
        expect(spy).toHaveReturnedWith({ color: '#636D75', icon: wrapper.vm.UnreachableIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor('');
        expect(spy).toHaveReturnedWith({ color: '#636D75', icon: wrapper.vm.UnreachableIcon });
        spy.mockRestore();
    });
    it('changeColor toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeColor');
        await wrapper.vm.changeColor('talking&calling');
        expect(spy).toHaveReturnedWith({ color: '#78ABCE', icon: wrapper.vm.CallingIcon });
        spy.mockRestore();
    });
    it('mounted component', async () => {
        await AgentsMap.mounted.call(wrapper.vm);
    });
    it('beforeDestroy component', async () => {
        await AgentsMap.beforeDestroy.call(wrapper.vm);
    });
});