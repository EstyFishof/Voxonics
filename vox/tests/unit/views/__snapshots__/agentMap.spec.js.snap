// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`support Check snapshot 1`] = `
<section
  class="content-wrapp"
>
  <div
    class="toolbar"
  >
    <div>
      <div
        class="btn-img"
      >
        <svg
          class="switch-view switch-list"
          content="List mode"
          fill="none"
          height="16"
          tabindex="0"
          viewBox="0 0 19 15"
          width="23"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="1.76923"
            rx="0.884615"
            width="14.2305"
            x="4.76953"
            y="0.5"
          />
           
          <rect
            height="1.76923"
            rx="0.884615"
            width="14.2305"
            x="4.76953"
            y="6.26953"
          />
           
          <rect
            height="1.76923"
            rx="0.884615"
            width="14.2305"
            x="4.76953"
            y="12.0391"
          />
           
          <circle
            cx="1.5"
            cy="1.5"
            r="1.5"
          />
           
          <circle
            cx="1.5"
            cy="7.19922"
            r="1.5"
          />
           
          <circle
            cx="1.5"
            cy="13"
            r="1.5"
          />
        </svg>
      </div>
       
      <div
        class="top-btn"
      >
        <div
          class="edit-map"
        >
          <img
            content="Edit mode"
            src="@/assets/img/icons/edit-map.svg"
            tabindex="0"
          />
        </div>
         
        <div
          class="mini-card "
        >
          <img
            content="Reset"
            src="@/assets/img/icons/to-back-map.svg"
            tabindex="0"
          />
        </div>
         
        <!---->
         
        <!---->
      </div>
       
      <ul
        class="tabs-sort"
      >
        <li
          class="toolbar-text"
        >
          Sort by:
        </li>
         
        <li>
          <label
            class="options-label"
          >
            <input
              class="input-field num"
              readonly="readonly"
              type="text"
            />
             
            <span
              class="triangle"
            >
              <svg
                fill="none"
                height="7"
                viewBox="0 0 12 7"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 1L6 6L11 1"
                  opacity="0.9"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </span>
             
            <ul
              class="list-related-numbers"
              style="display: none;"
            >
              <li>
                Name
              </li>
               
              <li>
                Agent number
              </li>
               
              <li>
                Location
              </li>
               
              <li>
                Time in status
              </li>
               
              <li>
                Talking to
              </li>
            </ul>
          </label>
        </li>
      </ul>
    </div>
  </div>
   
  <!---->
   
  <div
    class="drag-row"
  >
    <div
      style="width: 100%;"
    >
      <div
        class="list"
      >
        <drop-list-stub
          acceptsdata="function () { return true; }"
          acceptstype="function () { return true; }"
          class=""
          dragimageopacity="0.7"
          items=""
          mode="cut"
          tag="div"
        />
      </div>
    </div>
  </div>
</section>
`;
